source/connection.c

```diff
@@ -124,7 +135,24 @@ struct aws_http_connection *aws_http_connection_new_channel_handler(
         struct aws_channel_handler *tls_handler = tls_slot->handler;
         struct aws_byte_buf protocol = aws_tls_handler_protocol(tls_handler);
         if (protocol.len) {
-            if (aws_string_eq_byte_buf(s_alpn_protocol_http_1_1, &protocol)) {
+            bool customized = false;
+            if (alpn_string_map) {
+                struct aws_string *negotiated_result = aws_string_new_from_buf(alloc, &protocol);
+                struct aws_hash_element *found = NULL;
+                aws_hash_table_find(alpn_string_map, (void *)negotiated_result, &found);
+                if (found) {
+                    version = (enum aws_http_version)found->value;
+                    customized = true;
```

Comment:
if `found` is `false` we're going to end up falling back to HTTP/1.1
probably want it to be an error if `found` is false